nixos/modules/services/networking/iwd.nix

```diff
@@ -4,16 +4,141 @@ with lib;
 
 let
   cfg = config.networking.wireless.iwd;
+
+  encodeFileName = pkgs.writeScript "encode" ''
+    #! ${pkgs.runtimeShell} -e
+    echo "=$(printf "$1" | ${pkgs.unixtools.xxd}/bin/xxd -pu)"
+  '';
+
+  # If a network consists of alphanumeric chars and `_` and `-`, the network will be stored
+  # as `/var/lib/iwd/<ssid>.<ext>`. In any other case a hex-encoded version of the SSID will
+  # be stored with an `=` as prefix.
+  encodeIfNeeded = file: ext: let
+    allowedChars = stringToCharacters "01234556789abcdefghijklmnopqrstuvwxyz-_";
+    skipEncoding = all (char: elem (toLower char) allowedChars) (stringToCharacters file);
+  in
+    "/var/lib/iwd/${if skipEncoding then file else "$(${encodeFileName} \"${file}\")"}.${ext}";
+
+  # Basically `types.str`, but the `check` function validates the string-length as well.
+  fixedStringLength = min: max: mkOptionType {
+    name = "str-with-length";
+    description = if min == max
+      then "string with exactly ${toString min} chars."
+      else "string which must have at least ${toString min} and at most ${toString max} chars.";
+    inherit (types.str) merge;
+    check = x: isString x && stringLength x >= min && stringLength x <= max;
+  };
```

Comment:
Wishlist: some sort of test to ensure this all works as expected--at least to catch complete breakage.  Are these details re:knownnetworks population something likely to stay the same? I suspect so, but interested if anyone has better information on this.

Reply:
Do you have a good suggestion how to test wireless stuff in a VM test? Otherwise I'm currently not sure how to implement this :/

Reply:
@dtzWill do you have an idea how to integration-test wifi connectivity? Or would it be sufficient from your PoV to check if the `known-networks` cmd detects the declaratively specified networks?

(the full integration-test may be helpful to test my additional patch btw since this testing it against new `iwd` versions always takes some time).